script.js

```diff
@@ -134,23 +134,27 @@ const renderTodoList = (todoListState, onDestroy) => {
     notifyStateUpdated();
   };
 
-  inputField.addEventListener('keydown', (e) => {
+  const escapeEmpty = () => inputField.value = "";
+
+  const inputActions = (e) => {
     if (e.key === "Enter") {
       createNewTodoItem();
-    } 
-    
+    }
     if (e.key === "Escape") {
-      inputField.value = "";
+      escapeEmpty();
     };
-  });
+  }
 
-  listButton.addEventListener('click', () => createNewTodoItem());
+  inputField.addEventListener('keydown', inputActions);
+
+  listButton.addEventListener('click', createNewTodoItem);
 
   deleteCardIcon.addEventListener('click', () => {
     cardContainer.remove();
+    inputField.removeEventListener('keydown', inputActions);
```

Comment:
Missed cleaning up from click listener added on the L:150 `listButton.addEventListener('click', createNewTodoItem);`